inspec/a2-api-integration/controls/iam_v2.rb

```diff
@@ -427,63 +428,207 @@
-        http_method: 'PUT',
-        request_body: {
-          name: updatedName,
-          password: "newpassword",
-          previous_password: "wrongagain"
-        }.to_json
-      )
-      expect(resp.http_status).to eq 400
+    TEST_USER = {
+        id: CUSTOM_USER_ID,
+        name: "display name !#$#",
+        password: "chefautomate"
+    }
+
+    describe "when the user is not yet created" do
+      CREATED_ID = "unique-user-id"
+
+      after(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CREATED_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      describe "POST /iam/v2beta/users/:id" do
+        it "creates the user" do
+          createdName =  "i created my own name"
+          resp = automate_api_request("/apis/iam/v2beta/users",
+            http_method: 'POST',
+            request_body: {
+              id: CREATED_ID,
+              name: createdName,
+              password: 'something-new'
+            }.to_json
+          )
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:user][:name]).to eq createdName
+          expect(resp.parsed_response_body[:user][:id]).to eq CREATED_ID
+        end
+      end
+    end
+
+    describe "when multiple users exists" do
+      TEST_USER_ID_2 = 'inspec-user-2'
+      TEST_USER_2 = {
+        id: TEST_USER_ID_2,
+        name: "display name 2",
+        password: "chefautomate"
+      }
+
+      before(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users",
+          http_method: 'POST',
+          request_body: TEST_USER.to_json
+        )
+        expect(resp.http_status).to eq 200
+
+        resp = automate_api_request("/apis/iam/v2beta/users",
+          http_method: 'POST',
+          request_body: TEST_USER_2.to_json
+        )
+        expect(resp.http_status).to eq 200
+      end
+
+      after(:each) do
+        resp = automate_api_request("/apis/iam/v2beta/users/#{CUSTOM_USER_ID}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+
+        resp = automate_api_request("/apis/iam/v2beta/users/#{TEST_USER_ID_2}", http_method: 'DELETE')
+        expect(resp.http_status.to_s).to match(/200|404/)
+      end
+
+      describe "GET /iam/v2beta/users/" do
+        it "returns the list of users" do
+          resp = automate_api_request("/apis/iam/v2beta/users")
+          expect(resp.http_status).to eq 200
+          expect(resp.parsed_response_body[:users].length).to eq 3
+          expect(resp.parsed_response_body[:users][0][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
+          expect(resp.parsed_response_body[:users][1][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
+          expect(resp.parsed_response_body[:users][2][:id]).to match(/#{TEST_USER_ID_2}|#{CUSTOM_USER_ID}|#{ADMIN_USER_ID}/)
```

Comment:
That is not as strong as it could be; it would still pass if all three users in the response were the same user.
Here is one way to ensure that the response really contains all 3 users:
```
idList = resp.parsed_response_body[:users].map { |user| user.id }
expect((idList & [TEST_USER_ID_2, CUSTOM_USER_ID, ADMIN_USER]).size).to eq idList.size
```
Or, without introducing a temp variable:
```
resp.parsed_response_body[:users].map { |u| u.id }.all? 
    { |user| [TEST_USER_ID_2, CUSTOM_USER_ID, ADMIN_USER].include? user }
```

Reply:
I also think `match_array` might work, here. 

`expect(resp.parsed_response_body[:users].map(&:id)).to match_array([TEST_USER_ID_2, CUSTOM_USER_ID, ADMIN_USER])` 

sometimes that `&:id` helper is weird around json though

Reply:
(the one @msorens suggested, iterating over users, is better if we expect that there will be other users lurking around that our tests haven't created, which, sadly, has been true for some other things . . . tokens, I think)

Reply:
As long as we keep the length test in L497 then @blakestier 's is better--use the test helpers when available.